src/openrct2/object/Object.cpp

```diff
@@ -322,6 +322,29 @@ std::unique_ptr<IStream> ObjectAsset::GetStream() const
     return {};
 }
 
+std::string VersionString(Version version)
+{
+    return String::Format("%i.%i.%i", std::get<0>(version), std::get<1>(version), std::get<2>(version));
+}
+
+Version VersionTuple(std::string version)
+{
+    if (version.length() == 0)
+    {
+        return std::make_tuple(0, 0, 0);
+    }
+
+    auto nums = String::Split(version, ".");
+    uint16_t versions[3]{};
+
+    for (int i = 0; i < 3 && i < nums.size(); i++)
+    {
+        versions[i] = stoi(nums.at(i));
+    }
+
+    return std::make_tuple(versions[0], versions[1], versions[2]);
```

Comment:
```suggestion
    if (nums.size() != 3)
    {
        log_warning("Unexpected version string %s. Expected 'X.Y.Z'", version);
    }

    return { stoi(nums[0]), stoi(nums[1]), stoi(nums[2]) };
```

Reply:
None of the official objects use the X.Y.Z format, they use X.Y, and custom objects based on official objects do the same. The number of decades should be flexible.

Reply:
Hmm, it might still be worth warning if it's different from 2 or 3 then?

Reply:
It probably is worth it.

I think it might be better if the game left-adjusted the values e.g. 1.0 interprets as {0,1,0} instead of {1,0,0}.

Reply:
It is something to consider. Another thing is, can't we keep to just two numbers? "X.Y"

Reply:
It's possible, I think being forward-thinking to a 3-field number would be useful though.